services/api/routers/minio.py

```diff
@@ -120,8 +123,36 @@ async def post_discover_upload(file: UploadFile):
             raise e
     else:
         return { 'object_name': stat.object_name, 'etag': stat.etag }
-    # upload
+
+    # upload original size image
     upload = storage.put_object(crd.minio.bucket, object_name, file.file, length=-1, part_size=10*1024*1024)
+
+    # if file is an image, create and upload thumbnail image
+    supported_formats = {
+        'image/png':  'png',
+        'image/jpg':  'jpg',
+        'image/jpeg': 'jpeg',
+        'image/gif':  'gif'
+    }
+
+    if file.content_type in supported_formats:
+        file.file.seek(0)
+        thumbnail_size = (64, 64)
+        filename, _ = object_name.rsplit('.', 1)
+        image_format = supported_formats.get(file.content_type)
+
+        try:
+            content = await file.read()
+            image = Image.open(io.BytesIO(content))
+            image.thumbnail(thumbnail_size)
+            buffer = io.BytesIO()
+            image.save(buffer, format=image_format)
+            buffer.seek(0)
+            storage.put_object(crd.minio.bucket, f'{filename}_thumbnail.{image_format}', buffer, length=-1, part_size=10*1024*1024)
+        except Exception:
```

Comment:
Would you handle this in another way?

Reply:
I would handle it like this as well. Some remarks:

- `PIL` is not included in the Python standard distribution, you need to add it to `requirements.txt`
- extract thumbnail size and name extension to `api.config`
- `filename = object_name.rsplit('.', 1)[0]`
- It might be more efficient to chain the PIL methods, like:

  ```py
  buffer = io.BytesIO()
  Image.open(io.BytesIO(content)).\
      thumbnail(thumbnail_size).\
      save(buffer, format=image_format)
  buffer.seek(0)
  ```

- an alternative to the filename extension `_thumbnail` would be the thumbnail size, that can be read from the corresponding config value (`thumbnail_size[0]`)